react-useEffect/react-effects-and-fetch_iss-tracker/src/App.js

```diff
@@ -11,7 +11,19 @@ export default function App() {
     latitude: 0,
   });
 
-  async function getISSCoords() {}
+  async function getISSCoords() {
+    try {
+      const response = await fetch(URL);
+      const data = await response.json();
+      setCoords(data);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+  useEffect(() => {
+    let intervalId = setInterval(getISSCoords, 5000);
+    clearInterval(intervalId);
```

Comment:
Great solution, there is also another way of writing a clean up function just FYI

`useEffect(() => {
  let interValId = setInterval(getISSCoords, 5000);

  // Cleanup function
  return () => {
    // Clear the interval when the component unmounts or when the dependency array changes
    clearInterval(interValId);
  };
}, []);
`

Reply:
thanks for the feedback!

Reply:
Updated!